### PR TITLE
1313 dataset error routes

### PR DESCRIPTION
**Pull request by exbotanical**
The error boundary is redirected in the following cases:

1. Dataset does not exist
2. Dataset/draft is private and user lacks permissions
3. Dataset/draft is private and user has permissions, but a pending process is blocking visibility
4. Dataset is public but there is no snapshot 

Update: 
- elif cleanup / refactor checks
- add styles
- discrete FreshDesk component

Pending / Blocked:
- pull gql `getDataset` message to check against for case no. 2

### CODE REVIEW

**Comment by nellh** [comment]
FYI the error state is returned to the client already, but discarded by Apollo. Changing the query error policy to pass it to the receiving component will show that it's a permission denied error in the case the user lacks read or write access.

**Comment by exbotanical** [comment]
> FYI the error state is returned to the client already, but discarded by Apollo. Changing the query error policy to pass it to the receiving component will show that it's a permission denied error in the case the user lacks read or write access.

Thanks! This is exactly the issue I was working on :D

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/errors/errorBoundary.jsx]: We can remove this conditional. The loading state precedes any response, so if we wanted to include a spinner for it that would be the first condition to handle.

**Review by david-nishi — CHANGES_REQUESTED** [review]

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (exbotanical) [reply packages/openneuro-app/src/scripts/errors/errorBoundary.jsx]: Makes sense. Also - on second gander - I think the final else (return children) can be a standalone return statement; I've included this as well.

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/errors/errorBoundary.jsx]: we should reverse the order of assignment here to `this.state.error || this.props.error`

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/errors/errorBoundary.jsx]: Since loading occurs when the dataset is simply being fetched, the text might be changed to something like, "Dataset is currently being fetched." or "Fetching dataset..."

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/errors/errorBoundary.jsx:175]: nice!

**Review by david-nishi — CHANGES_REQUESTED** [review]

**Review by david-nishi — APPROVED** [review]

**Comment by nellh** [comment]
It looks like this branch no longer shares history with upstream for some duplicate commits. That should be resolved before merge in order to prevent adding duplicate commits to master.

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/errors/errorBoundary.jsx:129]: Message text is good, but I think users should be pointed to the freshdesk modal on this condition.

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/errors/errorBoundary.jsx]: message here is always falsy. error should just be the original error object (to be used by FreshdeskWidget).

File: packages/openneuro-app/src/scripts/errors/errorBoundary.jsx
Comment (david-nishi) [reply packages/openneuro-app/src/scripts/errors/errorBoundary.jsx:129]: 'This dataset has no associated snapshots.'

**Review by david-nishi — CHANGES_REQUESTED** [review]

**Review by david-nishi — APPROVED** [review]

**Pull request merged**
